### PR TITLE
MGN-775 ability to show past cashflows

### PR DESCRIPTION
**Pull request by pondmouse**
Showing past cahsflows, this will not include past rates, discount factors or amounts. 
Attached image show the same fixed/float swap with full and future cashflows
cashflows

Note the "n/a" comes from the nullabe Doubles in the formatter

### CODE REVIEW

**Comment by pondmouse** [comment]
Green build: https://ci.opengamma.com/job/Platform%20v2/job/OG-Platform%20Pull%20Requests/616/

File: projects/OG-Financial/src/main/java/com/opengamma/financial/analytics/model/fixedincome/CashFlowDetailsProvider.java [hunk 1]
```diff
@@ -69,28 +75,47 @@
...
+  @Deprecated
   public CashFlowDetailsProvider(MulticurveProviderInterface multicurveProviderInterface,
                                  ZonedDateTime zonedDateTime,
                                  SwapDefinition definition,
                                  InterestRateSwapSecurity security,
                                  PayReceiveType type) {
-    ArgumentChecker.notNull(multicurveProviderInterface, "multicurveProviderInterface");
-    ArgumentChecker.notNull(zonedDateTime, "zonedDateTime");
-    ArgumentChecker.notNull(definition, "definition");
-    ArgumentChecker.notNull(type, "type");
```
Comment (jacbop) [inline projects/OG-Financial/src/main/java/com/opengamma/financial/analytics/model/fixedincome/CashFlowDetailsProvider.java]: Are these ArgumentChecker valudations no longer useful? In the overloaded constructor we could still check for not null.

Comment (pondmouse) [inline projects/OG-Financial/src/main/java/com/opengamma/financial/analytics/model/fixedincome/CashFlowDetailsProvider.java]: They are checked all checked in the call down to the new constructor

Comment (jacbop) [inline projects/OG-Financial/src/main/java/com/opengamma/financial/analytics/model/fixedincome/CashFlowDetailsProvider.java]: Ah. In the assignment. Missed that.

**Pull request merged**